orchagent/mirrororch.cpp

```diff
@@ -75,10 +76,145 @@ MirrorOrch::MirrorOrch(TableConnector stateDbConnector, TableConnector confDbCon
     m_fdbOrch->attach(this);
 }
 
+bool MirrorOrch::bake()
+{
+    SWSS_LOG_ENTER();
+
+    // Freeze the route update during orchagent restoration
+    m_freezeUpdate = true;
+
+    deque<KeyOpFieldsValuesTuple> entries;
+    vector<string> keys;
+    m_mirrorTable.getKeys(keys);
+    for (const auto &key : keys)
+    {
+        vector<FieldValueTuple> tuples;
+        m_mirrorTable.get(key, tuples);
+
+        bool active = false;
+        string monitor_port;
+        string vlan_id;
+
+        for (const auto &tuple : tuples)
+        {
+            if (fvField(tuple) == MIRROR_SESSION_STATUS)
+            {
+                active = fvValue(tuple) == MIRROR_SESSION_STATUS_ACTIVE;
+            }
+            else if (fvField(tuple) == MIRROR_SESSION_MONITOR_PORT)
+            {
+                monitor_port = fvValue(tuple);
+            }
+            else if (fvField(tuple) == MIRROR_SESSION_VLAN_ID)
+            {
+                vlan_id = fvValue(tuple);
+            }
+        }
+
+        if (!active)
+        {
+            continue;
+        }
+
+        SWSS_LOG_NOTICE("Found mirror session %s active before warm reboot");
+
+        // Recover saved active session's monitor port
+        m_recoverySessionPortMap.emplace(key, vlan_id + ":" + monitor_port);
+    }
+
+    return Orch::bake();
+}
+
+bool MirrorOrch::postBake()
+{
+    SWSS_LOG_ENTER();
+
+    SWSS_LOG_NOTICE("Start MirrorOrch post-baking");
+
+    for (auto it = m_syncdMirrors.begin(); it != m_syncdMirrors.end(); ++it)
+    {
+        const string name = it->first;
+        auto& session = it->second;
+
+        if (m_recoverySessionPortMap.find(name) != m_recoverySessionPortMap.end())
+        {
+            auto tokens = tokenize(m_recoverySessionPortMap[name], ':');
+            string vlan_id = tokens[0];
+            string alias = tokens[1];
+
+            SWSS_LOG_NOTICE("Recover mirror session %s with monitor port %s",
+                    name.c_str(), alias.c_str());
+
+            Port port;
+            m_portsOrch->getPort(alias, port);
+
+            // If the port belongs to a VLAN
+            if (vlan_id != "0")
+            {
+                Port vlan;
+                m_portsOrch->getPort("Vlan" + vlan_id, vlan);
+
+                NeighborEntry neighbor_entry;
+                MacAddress mac;
+                m_neighOrch->getNeighborEntry(vlan.m_alias, neighbor_entry, mac);
+
+                SWSS_LOG_NOTICE("Monitor port %s belongs to neighbor %s with IP %s",
+                        port.m_alias.c_str(), vlan.m_alias.c_str(),
+                        neighbor_entry.ip_address.to_string().c_str());
+
+                session.nexthopInfo.nexthop = neighbor_entry.ip_address;
+            }
+            // If the port belongs to a LAG
+            if (port.m_lag_id)
```

Comment:
a lag port can also belong to a vlan, in this case, you will set the nexthop twice, once in vlan, once in lag.

Reply:
currently, the whole MirrorOrch does not support LAG in VLAN scenario. In order to have it supported, a separate pull request will be desired.